reference/day_procedures.py

```diff
@@ -0,0 +1,158 @@
+
+    :param spark: The spark session to use
+    :type spark: SparkSession
+    :param minimum_total: what is the minimum number of procedures in total that must be performed,
+        defaults to 100
+    :type minimum_total: int, optional
+    :param p_value: what p-value to use when we run the binomial tests, defaults to 0.001
+    :type p_value: float, optional
+    :return: a dictionary containing the types and their code lists
+    :rtype: dict[str, list[str]]
+    """
+    providers = (
+        spark.read.table("strategyunit.reference.ods_trusts")
+        .filter(F.col("org_type").startswith("ACUTE"))
+        .select(F.col("org_to").alias("provider"))
+        .distinct()
+        .persist()
+    )
+
+    fyear_criteria = F.col("fyear") == 201920
+
+    op = (
+        # TODO: replicating logic from outpatients. not DRY.
+        # but, cannot use nhp.raw_data.opa as this table needs to created before
+        read_data_with_provider(spark, "hes.silver.opa")
+        .filter(F.col("sex").isin(["1", "2"]))
+        .filter(F.col("apptage").isNotNull())
+        # end of todo
+        .filter(F.col("atentype").isin(["1", "2"]))  # only include F2F appointments
+        .filter(
+            ~F.col("sushrg").rlike("^(WF|U)")
+        )  # only include appointments with a procedure
+        .join(providers, "provider", "semi")
+        .filter(fyear_criteria)
+    )
+
+    df_op = (
+        _get_procedures(spark, "opa")
+        .join(op, on=["fyear", "procode3", "attendkey"], how="semi")
+        .groupBy("procedure_code")
+        .agg(F.count("attendkey").alias("op"))
+    )
+
+    ip = (
+        hes_apc.join(providers, "provider", "semi")
+        .filter(fyear_criteria)
+        .filter(F.col("classpat").isin("1", "2"))
+        .filter(F.col("admimeth").startswith("1"))
+    )
+
+    df_ip = (
+        _get_procedures(spark, "apc")
+        .join(ip, on=["fyear", "procode3", "epikey"], how="inner")
+        .withColumn(
+            "type", F.when(F.col("classpat") == "1", F.lit("ip")).otherwise("dc")
+        )
+        .groupBy("procedure_code")
+        .pivot("type")
+        .count()
+    )
+
+    df = (
+        df_ip.join(df_op, "procedure_code", how="outer")
+        .fillna(0)
+        .withColumn("total", F.col("dc") + F.col("ip") + F.col("op"))
+        .filter(F.col("total") >= minimum_total)
+        .toPandas()
+    )
+
+    df2 = (
+        df.melt(
+            id_vars=["procedure_code", "total"],
+            value_vars=["op", "dc"],
+            var_name="type",
+        )
+        .assign(
+            pu=lambda x: x.apply(
+                lambda y: binomtest(
+                    y.value, y.total, p=0.5, alternative="greater"
+                ).pvalue,
+                axis=1,
+            )
+        )
+        .assign(
+            po=lambda x: x.apply(
+                lambda y: binomtest(
+                    y.value, y.total, p=0.05, alternative="greater"
+                ).pvalue,
+                axis=1,
```

Comment:
Personally I find the variable names `df2`, with columns `po`, `pu` unclear.

Looks like the binomial test is using different default probabities, `0.05` and `0.5` is this intentional?

*Edit* OHH it's probability_usually, probability_occasionally. It took me a while to work this out.  Which is why you have the different values for p =?

Suggest you refactor variable names to make the intention of this test a little clearer, or consider adding a comment.